pydomcfg/domzgr/sco.py

```diff
@@ -0,0 +1,271 @@
+        Returns
+        -------
+        DataArray
+            xarray DataArray of the 2D envelope bathymetry
+        """
+
+        if self._rmax:
+
+            lsm = self._lsm
+
+            # set first land point adjacent to a wet cell to
+            # min_dep as this needs to be included in smoothing
+
+            # ------------------------------------------------------------
+            # This is the original NEMO Fortran90 code: translated
+            # in python it is very inefficient
+            # ------------------------------------------------------------
+            # zenv = depth.copy()
+            # env  = zenv.data
+            #
+            # nj = env.shape[0]
+            # ni = env.shape[1]
+            #
+            # for j in range(nj - 1):
+            #    for i in range(ni - 1):
+            #        if not lsm[j, i]:
+            #           ip1 = np.minimum(i + 1, ni)
+            #           jp1 = np.minimum(j + 1, nj)
+            #           im1 = np.maximum(i - 1, 0)
+            #           jm1 = np.maximum(j - 1, 0)
+            #           if (
+            #               depth[jp1, im1]
+            #               + depth[jp1, i]
+            #               + depth[jp1, ip1]
+            #               + depth[j, im1]
+            #               + depth[j, ip1]
+            #               + depth[jm1, im1]
+            #               + depth[jm1, i]
+            #               + depth[jm1, ip1]
+            #           ) > 0.0:
+            #               env[j, i] = min_dep
+            #
+            # zenv.data = env
+            # ------------------------------------------------------------
+
+            # ------------------------------------------------------------
+            # This is my translation into xarray. I think it does what
+            # it should, a part on the boundaries: I tested with AMM7,
+            # and zenv computed with NEMO-like code (above) and this
+            # one are perfectly identical apart for two single different
+            # points just on the border ... I don't think it will make
+            # a huge difference but if there is a better way to manage
+            # the borders with xarray and obtain exactly the same results
+            # of the original NEMO-like code, happy to use it.
+            # ------------------------------------------------------------
+            cst_lsm = lsm * 0.0
+            ngb_pnt = [-1, 0, 1]
+            for j, i in product(ngb_pnt, repeat=2):
+                if not (j == 0 and i == 0):
+                    lsm_sft = lsm.shift({lsm.dims[1]: i, lsm.dims[0]: j})
+                    cst_lsm += lsm_sft
+
+            cst_lsm = cst_lsm.where(lsm == 0, 0)
+            cst_lsm = cst_lsm.where(cst_lsm == 0, 1)
+            zenv = depth.where(cst_lsm == 0, self._min_dep)
+            for dim in lsm.dims:
+                for indx in [0, -1]:
+                    zenv[{dim: indx}] = depth[{dim: indx}]
```

Comment:
@oceandie I think this should work also at the boundaries. Can you try it with you bathy files please?
```suggestion
            cst_lsm = lsm.rolling({dim: 3 for dim in lsm.dims}, min_periods=2).sum()
            cst_lsm = cst_lsm.shift({dim: -1 for dim in lsm.dims})
            cst_lsm = (cst_lsm > 0) & (lsm == 0)
            zenv = depth.where(~cst_lsm, self._min_dep)
```

Reply:
@malmans2 works perfectly, thanks! I knew rolling was the way, I still need to get loops-thinking out of my head! :) btw, I didn't know how to use in xarray a double condition (since "where" can use only one condition right?), thanks!

Reply:
> I didn't know how to use in xarray a double condition (since "where" can use only one condition right?), thanks!

Yes, `where` only accepts one condition. I think the trick is to convert the condition to a bool so you can use stuff like `&` `|`.
e.g. `.where(da>1 & da < 2)`. Alternatively, you can use `np.logical_and/or`

Reply:
[`.isin`](http://xarray.pydata.org/en/stable/generated/xarray.DataArray.isin.html) is another good one for these kind of stuff...